packages/@aws-cdk/aws-cloudwatch/lib/contributor-insights-widget.ts

```diff
@@ -0,0 +1,170 @@
+import * as cdk from '@aws-cdk/core';
+import { LegendPosition } from './graph';
+import { ConcreteWidget } from './widget';
+/**
+ * The number of top contributors to show.
+ */
+export enum TopContributors {
+  /**
+   * The top 10 contributors
+   */
+  TOP10 = 10,
+  /**
+   * The top 25 contributors
+   */
+  TOP25 = 25,
+  /**
+   * The top 50 contributors
+   */
+  TOP50 = 50,
+  /**
+   * The top 100 contributors
+   */
+  TOP100 = 100,
+}
+
+/**
+ * Statistic to use over the aggregation period
+ */
+export enum OrderStatistic {
+  /**
+   * All values submitted for the matching metric added together.
+   * This statistic can be useful for determining the total volume of a metric.
+   */
+  SUM = 'Sum',
+  /**
+   * The highest value observed during the specified period.
+   * You can use this value to determine high volumes of activity for your application.
+   */
+  MAXIMUM = 'Maximum'
+}
+
+/**
+ * Properties for a Contributor Insights widget
+ */
+export interface ContributorInsightsWidgetProps {
+  /**
+   * Title for the widget
+   *
+   * @default No title
+   */
+  readonly title?: string;
+
+  /**
+   * Insights rule to display
+   */
+  readonly ruleName: string;
```

Comment:
Is an "insights rule" a resource that is created independently?

In that case, referencing it should follow the same rules as referencing other AWS resources: we don't accept buckets as `bucketName: string`, we accept bucket as `bucket: IBucket`.

By that same token, we're probably going to need an `IInsightRule` and an `InsightRule.fromInsightRuleName()` to produce one.

You could go as far as completely fill out the L2 for [`AWS::CloudWatch::InsightRule`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-cloudwatch-insightrule.html) but I will accept an `abstract` class that only allows importing existing insight rules as a minimal delta from what we have, to move this PR forward.